Signal/src/util/OWSBackupJob.h

```diff
@@ -11,10 +11,42 @@ extern NSString *const kOWSBackup_ManifestKey_EncryptionKey;
 extern NSString *const kOWSBackup_ManifestKey_RelativeFilePath;
 extern NSString *const kOWSBackup_ManifestKey_DataSize;
 
+@class OWSBackupIO;
+@class OWSBackupJob;
+@class OWSBackupManifestContents;
+
 typedef void (^OWSBackupJobBoolCompletion)(BOOL success);
 typedef void (^OWSBackupJobCompletion)(NSError *_Nullable error);
+typedef void (^OWSBackupJobManifestSuccess)(OWSBackupManifestContents *manifest);
+typedef void (^OWSBackupJobManifestFailure)(NSError *error);
 
-@class OWSBackupJob;
+@interface OWSBackupManifestItem : NSObject
+
+@property (nonatomic) NSString *recordName;
+
+@property (nonatomic) NSData *encryptionKey;
+
+// This property is only set for certain types of manifest item;
+@property (nonatomic, nullable) NSString *relativeFilePath;
+
+// This property is only set if the manifest item is downloaded.
+@property (nonatomic, nullable) NSString *downloadFilePath;
+
+// This property is only set if the manifest item is compressed.
```

Comment:
I assumed we were uniformly compressing everything. Which things aren't compressed?

Reply:
We currently only compress database snapshots.  I haven't bothered with:

* Manifest files (which are tiny).
* Attachments - which might benefit from this.  It's on the card.